### ISSUE TITLE
Run button needs to be replaced with Cancel button as soon as it's running

### ISSUE THREAD

**Issue by darobin** [w3c/specberus#149]
We need to maintain state there, because if you double-click on the Run button (which is guaranteed to happen whenever the UI is responding too slowly), you get the process started multiple times in parallel.
![lol](https://cloud.githubusercontent.com/assets/38491/6190580/2e44843a-b3a3-11e4-9f46-517927dcf794.png)

**Comment by astorije** [comment]
I personally really like this output. It shows that what's OK is *really* OK and that what's broken couldn't be more broken :-)

On a more serious note, it happens at double-click, and also when you share the URI: [this check](http://www.w3.org/2014/10/pubrules/?url=https%3A%2F%2Fw3c.github.io%2Fwebappsec%2Fspecs%2FCSP2%2Fpublished%2F2015-02-CR.html&profile=CR&validation=simple-validation&noRecTrack=false&informativeOnly=false&patentPolicy=pp2004&processDocument=2014) takes around 10 seconds to start on my side after you click the URI, and you really want to click the button when nothing is happening!

And for that reason, I think a disabled "Please Wait" button with a spinning wheel or something might be even easier than a Cancel button... or both!

**Comment by darobin** [comment]
The reason I suggested "Cancel" rather than "Please wait" is because if the process *really* gets stuck, with "Please wait" all you can do is reload the page (and possibly lose your info). With Cancel the process gets stopped, presumably the events are halted and all (yes, this requires a fair bit more code to do right), but it's better UI.

I'm saying this from experience: I used to work with JIRA and everyone would save their bug reports to a file before submitting because when it got stuck on a disabled "Please wait" you lost everything :)

Anyway, not super high priority but nice to have for UI.

**Comment by astorije** [comment]
Totally understandable, my point was just to underline that it also happens when opening such links.

**Comment by tripu** [comment]
Almost duplicated; a related issue was already reported.
Solving #63 would also solve this issue (but in a rudimentary way). I like @darobin's idea of offering the possibility of cancelling the process altogether, before you are given the option to run it again.
